File: app/src/main/java/cm/aptoide/pt/install/AppInstallerStatusReceiver.java
package cm.aptoide.pt.install;

import cm.aptoide.pt.packageinstaller.InstallStatus;
import rx.Observable;
import rx.subjects.PublishSubject;

public class AppInstallerStatusReceiver {

  private PublishSubject<InstallStatus> installStatusPublishSubject;

  public AppInstallerStatusReceiver(PublishSubject<InstallStatus> installStatusPublishSubject) {
    this.installStatusPublishSubject = installStatusPublishSubject;
  }

  public void onStatusReceived(InstallStatus installStatus) {
    installStatusPublishSubject.onNext(installStatus);
  }

  public Observable<InstallStatus> getInstallerInstallStatus() {
    return installStatusPublishSubject;
  }
}

File: aptoide-installer/src/main/java/cm/aptoide/pt/aptoide_installer/AppInstallerStatusReceiver.kt
package cm.aptoide.pt.aptoide_installer

import cm.aptoide.pt.packageinstaller.InstallStatus
import io.reactivex.Observable
import io.reactivex.subjects.PublishSubject

class AppInstallerStatusReceiver(private val installStatusPublishSubject: PublishSubject<InstallStatus>) {

  fun onStatusReceived(installStatus: InstallStatus) {
    installStatusPublishSubject.onNext(installStatus)
  }

  fun getInstallerInstallStatus(): Observable<InstallStatus> {
    return installStatusPublishSubject
  }

}